scripts/credential-manager/kubernetes-credential-manager.sh

```diff
@@ -1,13 +1,40 @@
 #!/bin/sh -xe
 
-WAIT_APPROVED_SEC=${WAIT_APPROVED_SEC:-600}
-CSR_NAME=${CSR_NAME:-nuvlaedge-csr}
+WAIT_APPROVED_SEC=${WAIT_APPROVED_SEC:-600}                                                                        
 
 SHARED="/srv/nuvlaedge/shared"
 SYNC_FILE=".tls"
 CA="/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
 USER="nuvla"
 
+echo "The NuvlaEdge UUID is set to: ${NUVLAEDGE_UUID}"
+echo "The MY_NAMESPACE is set to: ${MY_NAMESPACE}"
+env
+
+if [ ! -z $SET_MULTIPLE ]
+then
+    echo "We are in the multiple NE mode..."
+    # therefore need to add namespace to the CSR?
+    # and CRB naming
+    NAMESPACE=$(echo $NUVLAEDGE_UUID | awk -F "/" '{print $2}')
+    if [ -z $NAMESPACE ]
+    then
+        NAMESPACE=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 6; echo)
+        NAMESPACE="default"
+    fi
+    echo "The namespace string got set to: ${NAMESPACE}"
+    CSR_NAME=${CSR_NAME}-${NAMESPACE}
+    SYNC_FILE=".${NAMESPACE}.tls"
+    CRB_NAME=${USER}-crb-${NAMESPACE}
+else
+    CRB_NAME=${USER}-crb
+    SYNC_FILE=".tls"
+fi
+
+echo "The sync file is set to: $SYNC_FILE"                                         
+echo "The certificate siging request (CSR) is set to ${CSR_NAME}"  
```

Comment:
I don't think this change will work as expected because you removed the line 4 which was defining the default value (`nuvlaedge-csr`) for `CSR_NAME`; so if not defined externally, it will be `-${NAMESPACE}` (see line 26) if `SET_MULTIPLE` is defined and it will be not defined otherwise but it is used later on in the code.

Reply:
I put the CSR_NAME statement back in.